src/promptflow/promptflow/_sdk/operations/_run_operations.py

```diff
@@ -228,6 +228,31 @@ def update(
         ORMRun.get(name).update(display_name=display_name, description=description, tags=tags, **kwargs)
         return self.get(name)
 
+    @monitor_operation(activity_name="pf.runs.delete", activity_type=ActivityType.PUBLICAPI)
+    def delete(
+        self,
+        runs: Union[str, Run, List[str], List[Run]],
+        **kwargs,
+    ) -> None:
+        """Delete run permanently.
```

Comment:
what if the run is running? could we delete that?

Reply:
Yes, currently we don't have any guarantee, since we could not orchaestrate all running processes.
For process in use, if the running process hold the file, rmtree will fail directly.